crates/sparrow-runtime/src/read/parquet_file.rs

```diff
@@ -5,22 +5,21 @@ use arrow::record_batch::RecordBatch;
 use error_stack::{IntoReport, ResultExt};
 use futures::stream::BoxStream;
 use futures::{FutureExt, StreamExt, TryFutureExt};
-use object_store::ObjectStore;
+use object_store::{ObjectMeta, ObjectStore};
 use parquet::arrow::{
     parquet_to_arrow_schema_by_columns, ParquetRecordBatchStreamBuilder, ProjectionMask,
 };
 use parquet::errors::ParquetError;
 use parquet::file::metadata::ParquetMetaData;
-use parquet::file::FOOTER_SIZE;
 
 use crate::stores::{ObjectStoreRegistry, ObjectStoreUrl};
 
 /// A Parquet file in an [ObjectStore]. May be read asynchronously.
 #[derive(Clone)]
 pub struct ParquetFile {
     object_store: Arc<dyn ObjectStore>,
-    path: object_store::path::Path,
-    metadata: Arc<ParquetMetaData>,
+    pub object_meta: ObjectMeta,
```

Comment:
This makes it possible to get the `ObjectMeta` from the parquet file handle. Since we already made the `head` request, it doesn't make sense to not retain it.